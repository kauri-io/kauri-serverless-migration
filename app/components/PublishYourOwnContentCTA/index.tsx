import styled from 'styled-components'
import SecondaryButtonComponent from '../Button/SecondaryButton'
import { Title2 } from '../Typography'
import theme from '../../lib/theme-config'

const DEFAULT_CARD_WIDTH = theme.DEFAULT_CARD_WIDTH

const Container = styled.section`
    display: flex;
    flex-direction: column;
    > :first-child {
        margin-bottom: ${props => props.theme.space[2]}px;
    }
    @media (max-width: ${props => props.theme.breakpoints[0]}) {
        display: none;
    }
`

const Buttons = styled.div`
    display: flex;
    flex-direction: column;
    > :not(:last-child) {
        margin-bottom: ${props => props.theme.space[1]}px;
    }
`

interface IProps {
    content: Array<any | null> | null
    linkComponent: (
        children: React.ReactElement<any>,
        route: string
    ) => React.ReactElement<any>
    isLoggedIn: boolean
}

const PublishYourOwnContentCTA: React.FunctionComponent<IProps> = props => (
    <Container>
        <Title2>Publish your own content</Title2>
        <Buttons>
            {props.content &&
                props.content.map(
                    content =>
                        content &&
                        props.linkComponent(
                            <SecondaryButtonComponent
                                key={content.actionName}
                                color="textPrimary"
                                border={'primary'}
                                width={`${DEFAULT_CARD_WIDTH}px`}
                            >
                                {content.actionName}
                            </SecondaryButtonComponent>,
                            props.isLoggedIn
                                ? content.link
                                : `/login?r=${content.link}`
                        )
                )}
        </Buttons>
    </Container>
)

export default PublishYourOwnContentCTA
