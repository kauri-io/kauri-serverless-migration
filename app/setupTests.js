import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import theme from './lib/theme-config'
import { ThemeProvider } from 'styled-components'

Enzyme.configure({ adapter: new Adapter() })

const observeMock = {
    observe: () => null,
    disconnect: () => null, // maybe not needed
}

window.IntersectionObserver = () => observeMock

export const mountWithTheme = children =>
    mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const renderWithTheme = children =>
    render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const shallowWithTheme = children =>
    shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
