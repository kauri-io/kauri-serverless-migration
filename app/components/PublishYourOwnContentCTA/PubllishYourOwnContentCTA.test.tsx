import PublishYourOwnContentCTA from './index'
import { mountWithRedux } from '../../setupTests'

describe('components/PublishYourOwnContentCTA', () => {
    it('should match snapshot', () => {
        const PublishYourOwnContentCTAProps = {
            content: [],
            linkComponent: jest.fn(),
            isLoggedIn: true,
        }

        const wrapper = mountWithRedux(
            <PublishYourOwnContentCTA {...PublishYourOwnContentCTAProps} />
        )
        expect(wrapper).toMatchSnapshot()
    })
})
