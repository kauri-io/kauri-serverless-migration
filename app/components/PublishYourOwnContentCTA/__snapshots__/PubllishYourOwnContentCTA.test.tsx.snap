// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PublishYourOwnContentCTA should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "DEFAULT_CARD_WIDTH": 288,
        "MAIN_BODY_WIDTH": 1242,
        "bg": Object {
          "bgPrimary": "#1E2428",
          "bgSecondary": "#1E3D3B",
          "borderTextArea": "#979797",
          "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
          "disabledBackgroundColor": "#D6D6D6",
          "disabledTextColor": "#9B9B9B",
          "divider": "#EBEBEB",
          "errorRedColor": "#C03030",
          "hoverTextColor": "#108E72",
          "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
          "paddingTop": "2.5em",
          "primary": "#0BA986",
          "primaryColor": "#0BA986",
          "primaryDark": "#11856B",
          "primaryTextColor": "#1E2428",
          "secondary": "#11856B",
          "secondaryBlue": "rgb(81, 123, 162, 1)",
          "secondaryBlueDark": "rgb(81, 118, 142, 1)",
          "secondaryColor": "#1E3D3B",
          "secondaryTextColor": "#283035",
          "tertiaryBackgroundColor": "#F7F7F7",
          "textPrimary": "#1E2428",
          "white": "#FFFFFF",
        },
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "breakpoints": Array [
          "500px",
          "52em",
          "1242px",
        ],
        "colors": Object {
          "bgPrimary": "#1E2428",
          "bgSecondary": "#1E3D3B",
          "borderTextArea": "#979797",
          "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
          "disabledBackgroundColor": "#D6D6D6",
          "disabledTextColor": "#9B9B9B",
          "divider": "#EBEBEB",
          "errorRedColor": "#C03030",
          "hoverTextColor": "#108E72",
          "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
          "paddingTop": "2.5em",
          "primary": "#0BA986",
          "primaryColor": "#0BA986",
          "primaryDark": "#11856B",
          "primaryTextColor": "#1E2428",
          "secondary": "#11856B",
          "secondaryBlue": "rgb(81, 123, 162, 1)",
          "secondaryBlueDark": "rgb(81, 118, 142, 1)",
          "secondaryColor": "#1E3D3B",
          "secondaryTextColor": "#283035",
          "tertiaryBackgroundColor": "#F7F7F7",
          "textPrimary": "#1E2428",
          "white": "#FFFFFF",
        },
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "fontSizes": Array [
          11,
          13,
          14,
          16,
          18,
          20,
          22,
          28,
          32,
        ],
        "fontWeight": Array [
          300,
          "normal",
          500,
          700,
        ],
        "hoverTextColor": "#108E72",
        "padContent": "
    max-width: 1242px;
    margin-left: auto;
    margin-right: auto;
    @media (min-width: 1242px) {
      min-width: 1242px;
    }
    @media (max-width: 1242px) {
      min-width: 90%;
    }
  ",
        "padding": "calc((100vw - 1242px) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "space": Array [
          0,
          10,
          15,
          30,
          60,
        ],
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      }
    }
  >
    <PublishYourOwnContentCTA
      content={Array []}
      isLoggedIn={true}
      linkComponent={[MockFunction]}
    >
      <PublishYourOwnContentCTA__Container>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "PublishYourOwnContentCTA__Container-sc-1tajuth-0",
                "isStatic": false,
                "lastClassName": "bShpXn",
                "rules": Array [
                  "display:flex;flex-direction:column;>:first-child{margin-bottom:",
                  [Function],
                  "px;}@media (max-width:",
                  [Function],
                  "){display:none;}",
                ],
              },
              "displayName": "PublishYourOwnContentCTA__Container",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "PublishYourOwnContentCTA__Container-sc-1tajuth-0",
              "target": "section",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <section
            className="PublishYourOwnContentCTA__Container-sc-1tajuth-0 bShpXn"
          >
            <Typography__styledComponent>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "Typography__styledComponent-sc-36xkf6-5",
                      "isStatic": false,
                      "lastClassName": "eLGQne",
                      "rules": Array [
                        "color:",
                        [Function],
                        ";margin:0px;font-weight:",
                        "500",
                        ";",
                        ";font-size:",
                        "20",
                        "px;text-transform:",
                        [Function],
                        ";text-align:",
                        [Function],
                        ";:hover{color:",
                        [Function],
                        ";cursor:",
                        [Function],
                        ";}",
                      ],
                    },
                    "displayName": "Typography__styledComponent",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "Typography__styledComponent-sc-36xkf6-5",
                    "target": "span",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <span
                  className="Typography__styledComponent-sc-36xkf6-5 eLGQne"
                >
                  Publish your own content
                </span>
              </StyledComponent>
            </Typography__styledComponent>
            <PublishYourOwnContentCTA__Buttons>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "PublishYourOwnContentCTA__Buttons-sc-1tajuth-1",
                      "isStatic": false,
                      "lastClassName": "eqblHa",
                      "rules": Array [
                        "display:flex;flex-direction:column;>:not(:last-child){margin-bottom:",
                        [Function],
                        "px;}",
                      ],
                    },
                    "displayName": "PublishYourOwnContentCTA__Buttons",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "PublishYourOwnContentCTA__Buttons-sc-1tajuth-1",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <div
                  className="PublishYourOwnContentCTA__Buttons-sc-1tajuth-1 eqblHa"
                />
              </StyledComponent>
            </PublishYourOwnContentCTA__Buttons>
          </section>
        </StyledComponent>
      </PublishYourOwnContentCTA__Container>
    </PublishYourOwnContentCTA>
  </ThemeProvider>
</Provider>
`;
